buildchain/buildchain/lint.py

```diff
@@ -64,27 +75,86 @@ def lint_shell() -> types.TaskDict:
         shell_scripts.extend(constants.ROOT.glob('*/*{}'.format(ext)))
     return {
         'name': 'shell',
+        'title': _title,
         'doc': lint_shell.__doc__,
         'actions': [['tox', '-e', 'lint-shell']],
         'file_dep': shell_scripts,
     }
 
+# }}}
+# YAML {{{
 
 def lint_yaml() -> types.TaskDict:
     """Run YAML linting."""
     return {
         'name': 'yaml',
+        'title': _title,
         'doc': lint_yaml.__doc__,
         'actions': [['tox', '-e', 'lint-yaml']],
         'file_dep': list(constants.ROOT.glob('salt/**/*.yaml')),
     }
 
+# }}}
+# Go {{{
+
+def check_go_fmt() -> Optional[doit.exceptions.TaskError]:
+    """Check if Go code is properly formatted."""
+    cwd  = constants.STORAGE_OPERATOR_ROOT
+    cmd  = [
+        config.ExtCommand.GOFMT.value, '-s', '-d',
+        *tuple(constants.STORAGE_OPERATOR_FMT_ARGS)
+    ]
+    diff = subprocess.check_output(cmd, cwd=cwd)
+    if diff:
+        return doit.exceptions.TaskError(
+            msg='badly formatted Go code, please run `doit.sh format:go`'
+        )
+    return None
+
+
+def check_go_codegen() -> Optional[doit.exceptions.TaskError]:
+    """Check if the generated files are up to date."""
+    cwd  = constants.STORAGE_OPERATOR_ROOT
+    git_diff = [config.ExtCommand.GIT.value, 'diff']
+    base = subprocess.check_output(git_diff)
+    for target in ('k8s', 'openapi'):
+        cmd = [config.ExtCommand.OPERATOR_SDK.value, 'generate', target]
+        subprocess.check_call(cmd, cwd=cwd)
+    current = subprocess.check_output(git_diff)
+    # If the diff changed after running the code generation that means that
+    # the generated
+    if current != base:
```

Comment:
Should we reset the changes in that situation?

Reply:
If we reset, we could lose the existing diff of the user.

The buildchain should avoid to mess with the git state as much as possible (yes, here we are modifying files, but anyway the guy is gonna run `./doit.sh codegen` so it's not a big deal).